proxy/coco/src/request/waiting_room.rs

```diff
@@ -401,7 +401,8 @@ impl<T, D> WaitingRoom<T, D> {
             .filter_by_state(RequestState::Requested)
             .find(move |(_, request)| {
                 request.elapsed(timestamp.clone()) >= self.config.delta.clone()
-                    || request.attempts().clones == Clones::new(0)
+                    || (request.attempts().clones == Clones::new(0)
```

Comment:
Wait, I was thinking there wouldn't be _any_ other checks here. I believe `queries` will always be `> 0` because once you move the `Created` into `Requested` by calling `queried` it will bump the value.
```suggestion
```